pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java

```diff
@@ -771,8 +782,13 @@ private CompletableFuture<Void> filterAndUnloadMatchedNamespaceAsync(NamespaceIs
                             .map(tenant -> adminClient.namespaces().getNamespacesAsync(tenant));
                     return FutureUtil.waitForAll(completableFutureStream)
                             .thenApply(namespaces -> {
-                                // if namespace match any policy regex, add it to ns list to be unload.
+                                // Filter namespaces that have current cluster in their replication_clusters
+                                // if namespace match any policy regex, add it to ns list to be unloaded.
                                 return namespaces.stream()
+                                        .filter(namespaceName -> adminClient.namespaces()
+                                                .getPoliciesAsync(namespaceName)
+                                                .thenApply(policies -> policies.replication_clusters.contains(cluster))
+                                                .join())
```

Comment:
this would be a blocking operation. it would be better to make it asynchronous.

Reply:
For the filter to work, I need to call join to wait for the future completion. The reason for doing it this way is that policy is required to decide whether to remove the namespace or not. Can you suggest how I can make it async? Nothing better comes to my mind that doesn't need too much code refactoring...